### ISSUE TITLE
valid_ibinspectable: it always treat as invalid when we use type inference.

### ISSUE THREAD

**Issue by soranoba** [realm/SwiftLint#3326]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

It is correct.
```swift
class Foo {
    @IBInspectable var color: UIColor = UIColor.white
}
```

However, it is **NOT** correct.
```swift
class Foo {
    @IBInspectable var color = UIColor.white
}
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?: 0.40.1

* Installation method used (Homebrew, CocoaPods, building from source, etc)?: Homebrew

* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?: No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?: 11.6
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
let foo = try! bar()
```

**Comment by soranoba** [comment]
Since the typeName is not taken, it maybe that it is an issue of Source Kitten.

**Comment by soranoba** [comment]
Ahh.... Sorry.
This is my mistake.
When using this rule, we needs to type explicit.
